meinberlin/apps/projects/tasks.py

```diff
@@ -0,0 +1,107 @@
+from datetime import datetime
+from datetime import timedelta
+
+from celery import shared_task
+from django.core.cache import cache
+from django.utils import timezone
+
+from adhocracy4.phases.models import Phase
+from meinberlin.apps import logger
+
+
+def get_next_project_starts() -> str:
+    """
+    Helper function to query the db and
+    retreive the erliest phase that will start.
+    """
+    phase = (
+        Phase.objects.filter(start_date__isnull=False, start_date__gt=timezone.now())
+        .order_by("start_date")
+        .first()
+    )
+    if not phase:
+        return logger.warning("At least one phase with dates is required")
+    return phase.start_date.strftime("%Y-%m-%d, %H:%M:%S")
+
+
+def get_next_project_ends() -> str:
+    """
+    Helper function to query the db and
+    retreive the erliest phase that will end.
+    """
+    phase = (
+        Phase.objects.filter(end_date__isnull=False, end_date__gt=timezone.now())
+        .order_by("end_date")
+        .first()
+    )
+    if not phase:
+        return logger.warning("At least one phase with dates is required")
+    return phase.end_date.strftime("%Y-%m-%d, %H:%M:%S")
+
+
+@shared_task(name="schedule_reset_cache_for_projects_becoming_active")
+def schedule_reset_cache_for_projects_becoming_active():
+    """The task is set via celery beat every 10 minutes in
+    settings/production.txt.
+
+    Returns:
+        A string message if the next project starts
+        date will be removed from the cache.
+    """
+
+    # check if redis has a next_project_starts key
+    next_project_starts = cache.get("next_project_starts")
+
+    if not next_project_starts:
+        # set the redis key: value
+        cache.set("next_project_starts", get_next_project_starts())
```

Comment:
do we need to set `next_project_starts` here? otherwise it will be `None` below

Reply:
line 57 does that, setting the 'next_project_starts'.